WordPress/src/test/java/org/wordpress/android/ui/prefs/categories/CategoriesListViewModelTest.kt

```diff
@@ -1,25 +1,134 @@
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import junit.framework.TestCase.assertTrue
+import kotlinx.coroutines.InternalCoroutinesApi
 import org.junit.Before
-import org.junit.Rule
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.TEST_DISPATCHER
+import org.wordpress.android.fluxc.action.TaxonomyAction.FETCH_CATEGORIES
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.TaxonomyStore.OnTaxonomyChanged
+import org.wordpress.android.fluxc.store.TaxonomyStore.TaxonomyError
+import org.wordpress.android.fluxc.store.TaxonomyStore.TaxonomyErrorType.GENERIC_ERROR
+import org.wordpress.android.ui.posts.GetCategoriesUseCase
+import org.wordpress.android.ui.prefs.categories.CategoriesListViewModel.UiState
+import org.wordpress.android.ui.prefs.categories.CategoriesListViewModel.UiState.Content
+import org.wordpress.android.ui.prefs.categories.CategoriesListViewModel.UiState.Error.GenericError
+import org.wordpress.android.ui.prefs.categories.CategoriesListViewModel.UiState.Error.NoConnection
+import org.wordpress.android.ui.prefs.categories.CategoriesListViewModel.UiState.Loading
+import org.wordpress.android.util.NetworkUtilsWrapper
 
-@RunWith(MockitoJUnitRunner::class)
-class CategoriesListViewModelTest {
-    @Rule
-    @JvmField val rule = InstantTaskExecutorRule()
+@InternalCoroutinesApi
+class CategoriesListViewModelTest : BaseUnitTest() {
+    private val getCategoriesUseCase: GetCategoriesUseCase = mock()
+    private val networkUtilsWrapper: NetworkUtilsWrapper = mock()
+    private val siteModel: SiteModel = mock()
 
     private lateinit var viewModel: CategoriesListViewModel
 
+    private val uiStates = mutableListOf<UiState>()
+
     @Before
     fun setUp() {
-        viewModel = CategoriesListViewModel()
+        viewModel = CategoriesListViewModel(
+                getCategoriesUseCase,
+                networkUtilsWrapper,
+                TEST_DISPATCHER,
+        )
+        viewModel.uiState.observeForever { if (it != null) uiStates += it }
+    }
+
+    @Test
+    fun `when vm starts, then screen displays loading screen state`() {
+        viewModel.start(siteModel)
+        assertTrue(uiStates.first() is Loading)
+    }
+
+    @Test
+    fun `when vm starts, then screen displays list of items from db`() {
+        whenever(getCategoriesUseCase.getSiteCategories(siteModel)).thenReturn(mock())
+        viewModel.start(siteModel)
+        assertTrue(uiStates.first() is Loading)
+        assertTrue(uiStates.last() is Content)
+        assertTrue(uiStates.count() == 2)
+    }
+
+    @Test
+    fun `when no items are present in db and no internet available then no connection error should be displayed`() {
```

Comment:
Thanks for this info and explanation @ashiagr . Great Suggestion 🙌 . I think this would improve the readability of the tests to a great extent . Will do the changes 👍

Reply:
@ashiagr Have updated the test statements to the format of `given-when-then` format in 6703478